docs/coding-conventions.md

```diff
@@ -0,0 +1,335 @@
+  likely to contain hidden bugs (e.g. control flow issues).
+
+### Bad example
+
+```go
+val, err := doSomething()
+if err == nil {
+    if val {
+        return doSomethingElse()
+    } else {
+        return doAnotherThing()
+    }
+} else {
+    return err
+}
+```
+
+### Good example
+
+```go
+val, err := doSomething()
+if err != nil {
+    return err
+}
+
+if val {
+    return doSomethingElse()
+}
+
+return doAnotherThing()
+```
+
+## Use the same coding style throughout a file
+
+- Having a consistent coding style thought a file is recommended because it
+  makes reviews easier.
+- In case a cleaner agreed style is suggested by new code, an attempt should
+  be made to adjust the existing code as a follow-up contribution.
+- When you encounter code that requires a larger cleanup do it in a separate
+  commit or PR.
+
+### Bad example
+
+Consider `checkAB` already exists and `checkBC` is added:
+
+```go
+func checkAB(in string) bool {
+    return in == "A" || in == "B"
+}
+
+func checkBC(in string) bool {
+    switch in {
+    case "B", "C":
+        return true
+    }
+
+    return false
+}
+```
+
+### Good example
+
+Using the same style when adding new code:
+
+```go
+func checkAB(in string) bool {
+    return in == "A" || in == "B"
+}
+
+func checkBC(in string) bool {
+    return in == "B" || in == "C"
+}
+```
+
+# Document values meaning through constants or variable names
+
+- Use constants or variables if you would repeat common values otherwise.
+- Prefer encapsulation of common values and the operation on them through
+  objects and methods.
+- Carefully consider if you should make a constant exported.
+
+### Bad example
+
+```go
+func getImportantAnnotation(obj metav1.ObjectMeta) string {
+    return obj.GetAnnotations()["kubevirt.io/my-annotation"]
+}
+
+func setImportantAnnotation(obj metav1.ObjectMeta, val string) {
+    obj.GetAnnotations()["kubevirt.io/my-annotation"] = val
+}
+```
+
+### Good example
+
+```go
+const annotationKey = "kubevirt.io/my-annotation"
+
+func getImportantAnnotation(obj metav1.ObjectMeta) string {
+    return obj.GetAnnotations()[annotationKey]
+}
+
+func setImportantAnnotation(obj metav1.ObjectMeta, val string) {
+    obj.GetAnnotations()[annotationKey] = val
+}
+```
+
+### Another good example (avoiding exposure of constants)
+
+```go
+const levelProfessional = "professional"
+
+type Programmer struct {
+    level string
+}
+
+func (p *Programmer) setProfessional () {
+    p.level = levelProfessional
+}
+
+func (p *Programmer) isProfessional () bool {
+    return p.level == levelProfessional
+}
+```
+
+# Uniform import order and naming
+
+- Use the following import order with one block per item:
+    - Golang standard libraries
+    - Ginkgo / Gomega imports (only in test files)
+    - Third-party libraries (attempt to group the packages by domain when
+      possible)
+    - Local packages
+- Use the following naming schemes for imports:
+    - `v1` for imports of `kubevirt.io/api`
```

Comment:
Did you mean `kubevirt.io/api/core/v1` here and `virtv1` instead of `v1`?

I'm not even sure `virtv1` is valid given it's the `core` API group, `kubevirtv1` would make more sense IMHO.

Reply:
We had previous discussion on this and agreed to move forward with `v1` for our API in our tree. Changing to `kubevirt.io/api/core/v1` is valid though, thanks.